kore.maude

```diff
@@ -220,18 +227,30 @@ Questions about this definition:
 - do we really need ~> as a K construct?  Can it be a label and thus the way K
   deal with evaluation contexts just a particular methodology and not an
   intrisic part of K ? Same question for =>.
+  Cosmin: Syntacticly at least, I would say "no" for both. From the implementation
+  perspective, this will simplify traversal code even further. 
 - rename .::K and .::KList to . ?  if we get rid of ~> as above, then we may
   only need .::KList.
+  Cosmin: I agree.
 - Should we allow empty productions instead of sort declarations?  A pro
   argument is that we have fewer syntactic constructs.  A cons argument is
   that a sort declaration may be regarded as a different entityr, with
   different kinds of attributes.
+  Cosmin: I don't quite follow.
 - Should we add constant AttributeKeys which have special semantics, like
   "strict", as constants?
+  Cosmin: we may want to discuss the attribute syntax separately.
 - Brackets and disappearing injections are the same thing, arent's they?  A
   production which only has one non-terminal and the production disappears
   from the AST.
+  Cosmin: Yes, I also think they can be seen as the same thing. That being
+          said, I would really like for injections to not dissapear. If something
+          is meant to dissapear, it should be declared as a subsort. And the
+          dissapearing injections (including subsorts) could be done via rules as
+          we discussed before.
 - Should #Label contain more structured information than just a String?
+  Cosmin: From my perspective, its semantics and any other information should live
+          with the definition (module), not within the label.
 
```

Comment:
So far, I kind of saw this as an implementation detail. Once "resolved" to a module, the information for the klabel is clear (modulo overloaded klabels). It's up to the particular backend to choose between looking up the information from the module every time, having a direct reference to the KLabel information, or doing something completely different (e.g., instantiates a different class).

I can also see the advantage is keeping the information attached to the KLabel in KORE already. It would allow us to drop all syntax productions from a definition and it would still be valid. But it will come with a space overhead. I'm not sure it is worth it.